lib/build.py

```diff
@@ -274,6 +268,215 @@ def build_arg_get(arg):
+   """Temporary(?) analog of “modify_tree_make” for the non-interactive version
+      of “modify” using a script. For the command line:
+
+         $ ch-image modify foo foo2 /path/to/script
+
+      this function produces the following parse tree
+
+         start
+            dockerfile
+               from_
+                  image_ref
+                     IMAGE_REF foo
+               copy
+                  copy_shell
+                     WORD /path/to/script WORD /ch/script.sh
+               run
+                  run_shell
+                     LINE_CHUNK /ch/script.sh
+      """
+   # Children of dockerfile tree
+   df_children = []
+   # Metadata attribute. We use this attribute in the “_pretty” method for our
+   # “Tree” class. Constructing a tree without specifying a “Meta” instance that
+   # has been given a “line” value will result in the attribute not being present,
+   # which causes an error when we try to access that attribute. Here we give the
+   # attribute a debug value of -1 to avoid said errors.
+   meta = lark.tree.Meta()
+   meta.line = -1
+   df_children.append(im.Tree(lark.Token('RULE', 'from_'),
+                      [im.Tree(lark.Token('RULE', 'image_ref'),
+                      [lark.Token('IMAGE_REF', str(src_img))], meta)], meta))
+   if (cli.shell is not None):
+      df_children.append(im.Tree(lark.Token('RULE', 'shell'),
+                         [lark.Token('STRING_QUOTED', '"%s"' % cli.shell),
+                         lark.Token('STRING_QUOTED', '"-c"')],meta))
+   df_children.append(im.Tree(lark.Token('RULE', 'copy'),
+                      [im.Tree(lark.Token('RULE', 'copy_shell'),[lark.Token('WORD', path),
+                      lark.Token('WORD', '/ch/script.sh')], meta)],meta))
+   df_children.append(im.Tree(lark.Token('RULE', 'run'),
+                      [im.Tree(lark.Token('RULE', 'run_shell'),
+                      [lark.Token('LINE_CHUNK', '/ch/script.sh')], meta)],meta))
+   return im.Tree(lark.Token('RULE', 'start'), [im.Tree(lark.Token('RULE','dockerfile'), df_children)], meta)
+
+# Traverse Lark parse tree and do what it says.
+#
+# We don’t actually care whether the tree is traversed breadth-first or
+# depth-first, but we *do* care that instruction nodes are visited in order.
+# Neither visit() nor visit_topdown() are documented as of 2020-06-11 [1], but
+# examining source code [2] shows that visit_topdown() uses
+# Tree.iter_trees_topdown(), which *is* documented to be in-order [3].
+#
+# This change seems to have been made in 0.8.6 (see PR #761); before then,
+# visit() was in order. Therefore, we call that instead, if visit_topdown() is
+# not present, to improve compatibility (see issue #792).
+#
+# [1]: https://lark-parser.readthedocs.io/en/latest/visitors/#visitors
+# [2]: https://github.com/lark-parser/lark/blob/445c8d4/lark/visitors.py#L211
+# [3]: https://lark-parser.readthedocs.io/en/latest/classes/#tree
+def traverse_parse_tree(tree):
```

Comment:
Most significant first?

```suggestion
def parse_tree_traverse(tree):
```